jpeg_benchmark.py

```diff
@@ -0,0 +1,175 @@
+                   max_value: "int | float",
+                   current_value: "int | float",) -> "tuple[int, int] | tuple[float, float]":
+    """
+    Updates the min and max values for a measurement.
+
+    Args:
+        min_value: previous minimum value
+        max_value: previous maximum value
+        current_value: currently measured value
+
+    Returns: (min_value, max_value)
+        min_value: new updated minimum recorded value
+        max_value: new updated maximum recorded value
+        
+        The intended output is something like [int, int] or [float, float],
+        but it is not guaranteed because the inputs could be a combination of int and float.
+        eg. could also be tuple[float, int]
+    """
+    if current_value < min_value:
+        min_value = current_value
+    if current_value > max_value:
+        max_value = current_value
+
+    return min_value, max_value
+
+def run():
+    OUTPUT_PATH.mkdir(parents=True, exist_ok=True)
+    print("\n--------------Loading Images----------------\n")
+    images = [Image.open(INPUT_PATH / f"{i}.png") for i in range(FRAME_COUNT)]
+    print("\n--------------Images Loaded----------------\n")
+
+    # Set up results dictionary
+    results = {
+        f"lossy_{quality}": {
+            f"stimulated_fps_{sfps}": {
+                TOTAL_TIME_MS: 0,
+                AVG_TIME_MS: 0,
+                TOTAL_SPACE_B: 0,
+                AVG_SPACE_B: 0,
+                FRAME_SPACE: int(FRAME_COUNT/REAL_FPS * sfps),
+                FRAME_DATA: [],
+            } for sfps in STIMULATED_FPS
+        } for quality in QUALITY_SETTINGS
+    }
+
+    test_begin = time.time()
+    print("Start time:", test_begin)
+
+    for quality in QUALITY_SETTINGS:
+        quality_directory = "lossless" if quality == 100 else f"loss_{quality}"
+        if not os.path.exists(f"./results/{quality_directory}"):
+            os.mkdir(f"./results/{quality_directory}")
+
+        for sfps in STIMULATED_FPS:
+            TOTAL_SPACE_B = 0
+            if not os.path.exists(f"./results/{quality_directory}/{sfps}fps"):
+                os.mkdir(f"./results/{quality_directory}/{sfps}fps")
+
+            for i in range(NUM_REPEATS):
+                test_result = {
+                    TOTAL_TIME_MS: 0,
+                    AVG_TIME_MS: 0,
+                    TOTAL_SPACE_B: 0,
+                    AVG_SPACE_B: 0,
+                    FRAME_COUNT: int(FRAME_COUNT/REAL_FPS * sfps),
+                    FRAME_DATA: []
+                }
+                frame_ratio = int(REAL_FPS/sfps)
+
+                for jpeg in range(FRAME_COUNT):
+                    FILE_NAME = f"{jpeg}.{ENCODING}"
+                    image = Image.fromarray(images[jpeg*frame_ratio][:, :, ::-1])
+
+                    # Encode the frame with specified settings and time
+                    gc.disable()
+                    start = time.time_ns()
+                    image.save(f"./results/{quality_directory}/{sfps}fps/{FILE_NAME}",
```

Comment:
Instead of saving the image to a file (which takes way too much storage, creating 300 images again), just save it to temporary buffer (just some bytes). Only save the one image per batch (the one specified in `FRAME_TO_SAVE`) (see main for example)